### DISCUSSION TITLE
Question about the installation

### DISCUSSION THREAD

**Discussion by JohnyR82** [mathoudebine/turing-smart-screen-python#553] [Q&A]
Greetings,

I've recently purchased a small 3,5" screen (i.e. one of those "Turing Smart Screens") and I've noticed that besides the "original" software (which is hopefully "safe", but not very user friendly and a little bit buggy), there is also this custom made Python project.

I've never used Python or any program from GitHub before, and I've tried to follow the "How to start" guide, but so far, I've barely made it to "Step 3". I've installed Python, I've installed "Git"; and I've downloaded the project itself - although I am not sure where to download it (to the Python folder, to the Git folder, to Documents, or it doesn't even matter?). But so far, I got stuck with "Python dependencies"- because there are about 16 of them in the text file; and when I started searching for them, I am still not sure where to install them.

Now, based on the previous "Step 1", the "Git" program should be used to "download some Python dependencies" - but I am not sure if all of them, or some of them; or how even to do it. If the solution is the "code" at the end of the "Step 3" (i.e. "python3 -m pip install -r requirements.txt" / "python.exe -m pip install -r requirements.txt"), I am not sure where to type this code; or how to create the "virtual enviroment", that is recomended.

I have to say, I may have been a bit naive, because when I found out about the project, I thought it was a simple "program"; something like "system-monitor.exe" (or .rar) that I could download and install like the "original" software. And then I discovered the "guide"; but as I've mentioned, I am a bit lost. So if anyone has a little bit more "fool-proof" way how to guide a complete "Python-noob" to those "dependencies", it would be greatly appreciated.. :)

Thank you in advance,

Best regards,
Jan

**Comment by c0ldJS** [comment]
> ... I've downloaded the project itself - although I am not sure where to download it...

You can download wherever you like and even unpack it to some other directory if you want. Whatever makes sense to you. Nothing gets installed per se; once you've extracted the files someplace, consider that your "installation".

> ... how to create the "virtual enviroment", that is recomended.

This step is a personal choice (running a virtual environment isn't required) and will depend upon the version and age of Python installed on your machine. If you have a recent(ish) install of Python 3 then you're in luck because there's a dedicated tool for making them (`venv`). If you're on 2 or an older version of 3 then the steps are different.

`venv` requires a full path to wherever you have your smart screen project files _plus_ an additional directory, which will become the name of your virtual environment. If you want the environment called `myscreen` then here are some loose examples to get you started:

```
# Linux:
$ python -m venv /path/to/turing/myscreen

# Windows:
> python -m venv c:\path\to\turing\myscreen
```

Salt to taste, of course. If you're on Windows and don't have Python pathed then you may need to navigate to the directory where Python lives and issue the `venv` command there.

If you choose to travel this route, you will need to remember to always enter the virtual environment before doing any Python related things with the smart screen, like using the project's provided theme configuration tool (`theme-editor.py`), the preview generator (`turing-preview-generator.py`), and so on. Continuing upon the previous example, you'd enter the virtual environment by:
- Starting a command line session
- Navigating to `/path/to/turing/`
- Running `myscreen/Scripts/activate`

You'll notice the prompt changes to reflect your environment, and you're ready to do whatever you want to do. When you're done, exit the environment by typing `deactivate` on its own.

There are some benefits to utilizing a virtual environment for this but, again, it isn't mandatory. If all of the stuff above sounds too daunting or like too heavy of a lift, and you don't typically use Python for other stuff, you can totally forego the virtual environment and still use the project.

> If the solution is the "code" at the end of the "Step 3" ... I am not sure where to type this code...

If you chose the virtual environment option then it'll be from there.

Otherwise, it depends on whether or not you've pathed Python on the machine. If you haven't, navigate to the directory that has Python and do it there (making sure to supply the full path to the requirements file). If you did, Python can be invoked anywhere so just navigate to where said requirements file is and have at it.

---

One quick note -- getting the project up, running, and appearing on-screen in the way you'd like does take a little bit of getting the hands dirty. Please don't let that discourage you from digging in and trying things out. You might learn something cool or pick up some new tools for your toolbox along the way.

It's quite possible that you will never touch the Git stuff in the time you spend with your smart screen, where it only remains available for the initial steps for settings things up the first time. That's totally fine. If, later, you do elect to dip the toes in the proverbial Git waters then this project already has you setup to start adventuring.

As for Python,... the amount of time you spend with it will ultimately come down to how deep you want to go. Even if you decide to make your own themes, most of that exists in YAML with tools and features provided in the project already. There's no need to rapidly become a Python mage seasoned in the dark arts; there may only be one or two things you'll eventually pick up and use on a regular basis.

**Comment by c0ldJS** [comment]
@JohnyR82: How are things going? Have you had time to revisit and try anything out?

**Reply by GotrekGurninsson** [reply]
I'd love some help if you can offer some.

I'm running my screen in a fairly locked down OS based off Buildroot (called Batocera and used for retro emultation)

I've still managed to setup a virtual python environment, download all the requirements and get the screen mostly running.  But I think I'm missing a library and not sure how to find which one.

Everything works except my GPU (Nvidia) is not being detected 

(screen) [User@BetterSarah /userdata/turing]# python main.py
11/19/2024 10:41:56 PM [INFO] Loading theme Never_forget from res/themes/Never_forget/theme.yaml
11/19/2024 10:41:56 PM [DEBUG] HW revision: A
11/19/2024 10:41:56 PM [DEBUG] Auto detected COM port: /dev/ttyACM0
11/19/2024 10:41:56 PM [DEBUG] Using Python 3.11.8 (main, Jul 31 2024, 20:06:15) [GCC 12.3.0]
11/19/2024 10:41:56 PM [WARNING] Tray icon is not supported on your platform
11/19/2024 10:41:56 PM [INFO] Display reset (COM port may change)...
11/19/2024 10:42:01 PM [DEBUG] Static COM port: /dev/ttyACM0
11/19/2024 10:42:02 PM [DEBUG] HW sub-revision: SubRevision.TURING_3_5
11/19/2024 10:42:02 PM [DEBUG] Drawing Image: BACKGROUND
11/19/2024 10:42:02 PM [DEBUG] Bitmap res/themes/Never_forget/background.png is now loaded in the cache
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_CPU
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_CPUTEMP
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_CPUTEMPUNIT
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_GPUTEMPUNIT
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_GPUFAN
11/19/2024 10:42:02 PM [DEBUG] Drawing Text: LABEL_GPUCORE
11/19/2024 10:42:02 PM [WARNING] No supported GPU found
11/19/2024 10:42:02 PM [WARNING] If you have an AMD GPU, you may need to install some  libraries manually: see https://github.com/mathoudebine/turing-smart-screen-python/wiki/Troubleshooting#linux--macos-no-supported-gpu-found-with-an-amd-gpu-and-python-311


however because I have Arch Linux setup in my Batocera build, if I run a terminal from the Arch distro and run my screen from there (form the same location, the GPU is detected fine so I know it can work I just need to find the missing link.  I'll take any advice!  :D  I'm learning tons and love it.

Log form running under Arch Linux (still within Emulation Station in Baotera)

11/19/2024 10:59:36 PM [DEBUG] HW revision: A
11/19/2024 10:59:36 PM [DEBUG] Auto detected COM port: /dev/ttyACM0
11/19/2024 10:59:36 PM [DEBUG] Using Python 3.12.7 (main, Oct  1 2024, 11:15:50) [GCC 14.2.1 20240910]
11/19/2024 10:59:36 PM [INFO] Tray icon has been displayed
11/19/2024 10:59:36 PM [INFO] Display reset (COM port may change)...
11/19/2024 10:59:36 PM [ERROR] SerialException: Failed to send serial data to device. Closing and reopening COM port before retrying once.
11/19/2024 10:59:37 PM [DEBUG] Static COM port: /dev/ttyACM0
11/19/2024 10:59:37 PM [ERROR] Cannot open COM port /dev/ttyACM0: [Errno 2] could not open port /dev/ttyACM0: [Errno 2] No such file or directory: '/dev/tt>
11/19/2024 10:59:41 PM [DEBUG] Static COM port: /dev/ttyACM0
11/19/2024 10:59:42 PM [DEBUG] HW sub-revision: SubRevision.TURING_3_5
11/19/2024 10:59:42 PM [DEBUG] Drawing Image: BACKGROUND
11/19/2024 10:59:42 PM [DEBUG] Bitmap res/themes/Never_forget/background.png is now loaded in the cache
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_CPU
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_CPUTEMP
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_CPUTEMPUNIT
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_GPUTEMPUNIT
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_GPUFAN
11/19/2024 10:59:42 PM [DEBUG] Drawing Text: LABEL_GPUCORE
11/19/2024 10:59:42 PM [INFO] Detected Nvidia GPU(s)

If I run the demo_GPUtil.py script it does not find the GPU either

**Comment by ATLGAMES** [comment]
I suspect he just wanted to use the code, not become a python developer. I've been in IT for 30 years and getting this software to run is virtually impossible. Half of the mentioned 'requirements' are not even available in my distros repositories. Hunting down and compiling those just to get it to run is beyond the scope of most people's casual interest.  It took me about 3 hours to track down why tkinter could not be found.

**Reply by GNKyrios** [reply]
> I suspect he just wanted to use the code, not become a python developer. I've been in IT for 30 years and getting this software to run is virtually impossible. Half of the mentioned 'requirements' are not even available in my distros repositories. Hunting down and compiling those just to get it to run is beyond the scope of most people's casual interest. It took me about 3 hours to track down why tkinter could not be found.

im stuck on that as well, tkinter is installed, i can open the module from IDLE, can even execute it for the regular test, but running configure.py always says tkinter is not found. Plus like you said some extensions or modules wont install from cmd from the requirements.txt file, i downloaded them manually by name, but still cant get pass the tkinter error.

What did you do?

**Reply by GNKyrios** [reply]
Ok i solved it, seems to be some issue with post 3.12 python version, it worked better with 3.11. So for anyone having problems first do that. Then you can download all the requirements with less problems

Then the "import tkinter.ttk as ttk" line doesnt actually import as an alias, since i read you cant do that with sys modules, so i had to modify that part of the code to this instead:
`import tkinter.ttk as ttk
    from tkinter import *`
just by doing that the code actually ran and i got the config screen, and then i had to fix the librehardwaremonitor thing.

Now to find if i cant export the custom design i made on the original software . (hadnt read the themes post, you cant, just the images)
